src/core/config_mod.F90

```diff
@@ -294,7 +294,7 @@ SUBROUTINE Config_Grid( Config, Config_Opt, State_Grid, RC )
 !
       TYPE(QFYAML_t),      INTENT(INOUT) :: Config      ! YAML Config object
       TYPE(OptConfig),     INTENT(INOUT) :: Config_Opt   ! Input options
```

Comment:
```suggestion
      TYPE(QFYAML_t),      INTENT(INOUT) :: ConfigInput      ! YAML Config object
      TYPE(OptConfig),     INTENT(INOUT) :: Config   ! Input options
```
Something like this makes more sense to me. The info from the YAML is input, whereas the second is the real config, the internal config container.